PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/WhatsNewHelper.java

```diff
@@ -0,0 +1,125 @@
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.testFramework.LightVirtualFile;
+import com.microsoft.azure.hdinsight.common.StreamUtil;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.intellij.plugins.markdown.ui.preview.MarkdownSplitEditor;
+import org.intellij.plugins.markdown.ui.split.SplitFileEditor;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public enum WhatsNewHelper {
+    INSTANCE;
+
+    private static final String AZURE_TOOLKIT_FOR_JAVA = "Azure Toolkit for Java";
+    private static final String AZURE_TOOLKIT_WHATS_NEW = "AzureToolkit.WhatsNew";
+    private static final String VERSION_PATTERN = "<!-- Version: (.*) -->";
+    private static final String WHAT_S_NEW_CONSTANT = "WHAT_S_NEW";
+    private static final String WHAT_S_NEW_CONTENT_PATH = "/whatsnew/whatsnew.md";
+    private static final Key<String> WHAT_S_NEW_ID = new Key<>(WHAT_S_NEW_CONSTANT);
+
+    public synchronized void showWhatsNew(boolean force, Project project) throws IOException {
+        final FileEditorManager fileEditorManager = FileEditorManager.getInstance(project);
+        final VirtualFile existingWhatsNewFile = searchExistingFile(fileEditorManager);
+        if (existingWhatsNewFile != null) {
+            fileEditorManager.openFile(existingWhatsNewFile, true, true);
+        } else {
+            // Get whats new file
+            final String content = getWhatsNewContent();
+            // Get whats new version
+            final DefaultArtifactVersion whatsNewVersion = getWhatsNewVersion(content);
+            final DefaultArtifactVersion shownVersion = getShownVersion();
+            if (force || !isDocumentShownBefore(whatsNewVersion, shownVersion)) {
+                saveShownVersion(whatsNewVersion);
+                createAndShowWhatsNew(project, fileEditorManager, content);
+            }
+        }
+    }
+
+    private void createAndShowWhatsNew(Project project, FileEditorManager fileEditorManager, String content)
+            throws IOException {
+        final LightVirtualFile virtualFile = new LightVirtualFile(AZURE_TOOLKIT_FOR_JAVA);
+        virtualFile.setLanguage(Language.findLanguageByID("Markdown"));
+        virtualFile.setBinaryContent(content.getBytes());
+        virtualFile.putUserData(WHAT_S_NEW_ID, WHAT_S_NEW_CONSTANT);
+        final FileEditor[] fileEditors = fileEditorManager.openFile(virtualFile, true, true);
+        for (FileEditor fileEditor : fileEditors) {
+            if (fileEditor instanceof MarkdownSplitEditor) {
+                // Switch to markdown preview panel
+                ((MarkdownSplitEditor) fileEditor).triggerLayoutChange(SplitFileEditor.SplitEditorLayout.SECOND,
```

Comment:
Shall we check user data for what's new editor here? I'm wonder if other Markdaow editor opened will be impacted or not.

Reply:
Changed to use our own file editor to show markdown notes, so should not break users experience.